udata/core/spatial/models.py

```diff
@@ -64,6 +87,17 @@ def keys_values(self):
                 keys_values.append(value)
         return keys_values
 
+    @property
+    def url(self):
+        return url_for('gouvfr.territory', territory=self)
+
+    @property
+    def external_url(self):
+        return '{scheme}://{domain}{path}'.format(
```

Comment:
`return url_for('gouvfr.territory', territory=self, _external=True)`
The domain and scheme will be automatically set.